components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/util/APIClientTest.java

```diff
@@ -452,7 +463,34 @@ public void testCallAPIRetryOnTimeoutAndReceiveSuccessResponse() throws Exceptio
         entity.setContentType(ContentType.APPLICATION_JSON.getMimeType());
         when(httpResponse.getEntity()).thenReturn(entity);
 
-        ActionInvocationResponse response = apiClient.callAPI("http://example.com", null, "{}");
+        ActionInvocationResponse response = apiClient.callAPI(ActionType.PRE_ISSUE_ACCESS_TOKEN,
+                "http://example.com", null, "{}");
+
+        assertNotNull(response);
+        assertTrue(response.isSuccess());
+        verify(httpClient, times(2)).execute(any(HttpPost.class));
+    }
+
+    @Test
+    public void testCallAPIRetryOnTimeoutAndReceiveSuccessResponseWithExtendedResponseData() throws Exception {
+
+        when(httpClient.execute(any(HttpPost.class))).thenThrow(new ConnectTimeoutException("Timeout"))
+                .thenReturn(httpResponse);
+        when(httpResponse.getStatusLine()).thenReturn(statusLine);
+        when(statusLine.getStatusCode()).thenReturn(200);
+
+        ActionInvocationResponseClassFactory.registerActionInvocationResponseClassProvider(
+                new TestActionInvocationResponseClassProvider());
+
+        String successResponse =
+                "{\"actionStatus\":\"SUCCESS\", \"data\": {\"id\":\"test-123-id\"}}";
+        InputStreamEntity entity = new InputStreamEntity(new ByteArrayInputStream(successResponse.getBytes(
+                StandardCharsets.UTF_8)));
+        entity.setContentType(ContentType.APPLICATION_JSON.getMimeType());
+        when(httpResponse.getEntity()).thenReturn(entity);
+
+        ActionInvocationResponse response = apiClient.callAPI(ActionType.AUTHENTICATION,
+                "http://example.com", null, "{}");
```

Comment:
Why this test case needs to be added. We just want to validate the success response with some extended data right. It's not related to retry check